src/components/sbb-menu/sbb-menu.tsx

```diff
@@ -0,0 +1,297 @@
+import {
+  Component,
+  ComponentInterface,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  JSX,
+  Listen,
+  Method,
+  Prop,
+  State,
+  Watch,
+} from '@stencil/core';
+import { getElementPosition } from '../../global/helpers/position';
+import { isBreakpoint } from '../../global/helpers/breakpoint';
+
+const MENU_OFFSET = 8;
+
+const INTERACTIVE_COMPONENTS = ['SBB-BUTTON', 'SBB-LINK', 'SBB-MENU-ACTION'];
+
+const IS_FOCUSABLE_QUERY = `
+  button:not([disabled]),
+  [href],
+  input:not([disabled]),
+  select:not([disabled]),
+  textarea:not([disabled]),
+  details:not([disabled]),
+  summary:not(:disabled),
+  [tabindex]:not([tabindex="-1"]):not([disabled]),
+  sbb-button:not([disabled]),
+  sbb-link:not([disabled]),
+  sbb-menu-action:not([disabled])`;
+
+/**
+ * @slot unnamed - Use this slot to project any content inside the dialog.
+ */
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-menu.scss',
+  tag: 'sbb-menu',
+})
+export class SbbMenu implements ComponentInterface {
+  @Element() public el!: HTMLElement;
+
+  /**
+   * The element that will trigger the menu dialog.
+   * Accepts both a string (id of an element) or an HTML element.
+   */
+  @Prop({ reflect: true }) public trigger: string | HTMLElement;
+
+  /**
+   * Whether the animation is enabled.
+   */
+  @Prop() public disableAnimation = false;
+
+  /**
+   * Whether the menu is open.
+   */
+  @State() private _open = false;
+
+  /**
+   * Whether the menu is closing.
+   */
+  @State() private _isDismissing = false;
+
+  /**
+   * Emits whenever the menu starts the opening transition.
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'sbb-menu_will-open',
+  })
+  public willOpen: EventEmitter<void>;
+
+  /**
+   * Emits whenever the menu is opened.
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'sbb-menu_did-open',
+  })
+  public didOpen: EventEmitter<void>;
+
+  /**
+   * Emits whenever the menu begins the closing transition.
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'sbb-menu_will-close',
+  })
+  public willClose: EventEmitter<void>;
+
+  /**
+   * Emits whenever the menu is closed.
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'sbb-menu_did-close',
+  })
+  public didClose: EventEmitter<void>;
+
+  /**
+   * Opens the menu on trigger click.
+   */
+  @Method()
+  public async open(): Promise<void> {
+    this.willOpen.emit();
+    this._setMenuPosition();
+    this._dialog.showModal();
+  }
+
+  /**
+   * Closes the menu.
+   */
+  @Method()
+  @Listen('sbb-menu-action_click')
+  public async close(): Promise<void> {
+    this.willClose.emit();
+    this._isDismissing = true;
+  }
+
+  // Resets position on window resize.
+  @Listen('resize', { target: 'window', passive: true })
```

Comment:
This causes the event listener to be bound to the window, even if the menu is not used.
I would probably prefer, if this is only done, when the menu is opened (and removed when closed/disconnected).
@dauriamarco, @jeripeierSBB Thoughts?
(This also applies to the keydown event.)

Reply:
True, I did as you suggested for the resize event. While for the keydown event I just removed the `target: 'window'` since it wasn't really needed.